README.md

```diff
@@ -17,62 +17,98 @@ As a part of the bridge, a Kafka producer will be responsible for producing mess
 
 ### Topic Mapping Rules (ToMaR)
 
-The ToMaR is a set of patterns the user provides defining how the MQTT Bridge maps MQTT topic names to Kafka topic names.
-A Mapping Rule is a model that contains an `MQTT topic pattern` and a `Kafka topic template`. 
-All the incoming MQTT message's topic should match an `MQTT topic pattern` in the ToMaR so that the bridge knows in which Kafka topic to produce this message.
+The ToMaR is a set of patterns the user provides defining how the MQTT Bridge maps MQTT topic names to Kafka topic
+names.
+A Mapping Rule is a model that contains an `MQTT topic pattern`, a `Kafka topic template`, and optionally
+a `Kafka record key`.
+All the incoming MQTT message's topic should match an `MQTT topic pattern` in the ToMaR so that the bridge knows in
```

Comment:
```suggestion
All the incoming MQTT message's topic should match an `MQTT topic pattern` in the ToMaR so that the bridge knows in which Kafka topic to produce this message.
```